src/main/scala/graphql/schema/RedshiftType.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic.graphql.schema
+
+import basic.misc.Redshift
+import basic.graphql.syntax.all._
+import io.circe.Json
+import io.circe.syntax._
+import sangria.ast._
+import sangria.schema._
+import sangria.validation.ValueCoercionViolation
+import sangria.marshalling.ToInput
+import sangria.marshalling.circe._
+
+object RedshiftType {
+
+  object CoercionViolation extends ValueCoercionViolation("Redshift must be a number.")
+
+  def scalar: ScalarType[Redshift] =
```

Comment:
This scalar type definitions in Sangria are awful. You just have to provide a bunch of cases for various coercions and there's no way to know what may be passed to you.